### DISCUSSION TITLE
Does DuckDB support Azure Data Lake Storage (ADLS) authentication using the Device Code Flow method?

### DISCUSSION THREAD

**Discussion by devbrunorm** [duckdb/duckdb#11213] [General]
I am trying to authenticate to an ADLS using Device Code Flow within DuckDB. I have checked the documentation and it seems that some methods are already supported, but it appears that using Azure Refresh Token is not yet implemented. Is this correct? If so, would it be easy to implement?

I ask this because my company's development environment uses device code authentication to ensure greater security (where each user has their own separate token). However, my team would like to use DuckDB in this type of environment for our solutions.

If I have misunderstood and there is currently a way to authenticate with this token, I would be extremely grateful if someone could teach me how to do it.

Thanks for your attention and availability.

**Comment by FelipeGaleao** [comment]
Hi, @devbrunorm... Nice! 

I also had this doubt.
I hope the community responds as soon as possible!

**Comment by quentingodeau** [comment]
Hello, 
Not yet for the moment only the [followings](https://duckdb.org/docs/extensions/azure.html#authentication) authentication are available (I try to keep the doc up to date with my PRs). But please feel free to create a ticket in the [duckdb_azure](https://github.com/duckdb/duckdb_azure/issues/20) repository. I will try to take a look

**Reply by devbrunorm** [reply]
Thank you very much for the response. I will create a ticket in the repository, as requested. I don't know much about C++, but I hope I can help in some way with the implementation of this feature.

**Reply by devbrunorm** [reply]
I just read the issue you provided. I might be mistaken, but I think it could possibly solve our problem. I confess that the various Microsoft authentication methods leave me a little confused, but if EntraID's case is different from ours, here's a link explaining how this authentication actually works:

https://learn.microsoft.com/en-us/entra/identity-platform/v2-oauth2-device-code

**Reply by quentingodeau** [reply]
My apologies I have put the wrong link... So yes the link issue is unrelated to yours (it was something that I was looking into)!
The link I wanted to put is the [duckdb_azure repo](https://github.com/duckdb/duckdb_azure) and just create a new issue with just the information and the link to this discussion.
Sorry again for the wrong copy&paste...
